packages/mermaid/src/rendering-util/rendering-elements/shapes.ts

```diff
@@ -56,8 +58,12 @@ import { waveEdgedRectangle } from './shapes/waveEdgedRectangle.js';
 import { waveRectangle } from './shapes/waveRectangle.js';
 import { windowPane } from './shapes/windowPane.js';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-type ShapeHandler = (parent: any, node: Node, options: ShapeRenderOptions) => unknown;
+type MaybePromise<T> = T | Promise<T>;
```

Comment:
nit: Might be useful to move to a more generic types file, as it might be useful in other places.

Reply:
I've seen this type used in a lot of places in other code-bases (e.g. https://github.com/sveltejs/kit/blob/ed6b5cd4fc0d1d299c838628b73027dd07179281/packages/kit/types/index.d.ts#L2151), so you're right! (although it's not part of [type-fest](https://github.com/sindresorhus/type-fest), maybe because it's too simple).

I've moved it to `packages/mermaid/src/types` in https://github.com/mermaid-js/mermaid/pull/5974/commits/f8746bee0454b01378726d04f9c65bbc3974eb3a:

https://github.com/mermaid-js/mermaid/blob/f8746bee0454b01378726d04f9c65bbc3974eb3a/packages/mermaid/src/types.ts#L98-L103